aaelf64/aaelf64.rst

```diff
@@ -1260,16 +1260,32 @@ Linkers may optionally optimize instructions affected by relocation. Relocation
 
     ::
 
-      ADD x0, x1, 0   // eg. R_<CLS>_TLSLE_ ADD_TPREL_HI12
-      ADD x2, x2, 0   // or R_<CLS>_ADD_ABS_LO12_NC
+      ADD   x0, x1, 0   // eg. R_<CLS>_TLSLE_ADD_TPREL_HI12
+      ADD   x2, x2, 0   // or R_<CLS>_ADD_ABS_LO12_NC
 
       // after optimization:
 
-      MOV x0, x1
+      MOV   x0, x1
       NOP
 
   - The relocation ``R_<CLS>_ADR_PREL_PG_HI21`` may emit a MOV with zero immediate for undefined weak symbols.
 
+  - The following TLS relocations may be optimized if the symbol is a non-preemtable definition and the TLS offset fits in 16 bits:
+
+    ::
+
+      ADRP  x0, :gottprel: symbol            // R_<CLS>_TLSIE_ADR_GOTTPREL_PAGE21
+      LDR   x1, [x0, :gottprel_lo12: symbol] // R_<CLS>_TLSIE_LD64_GOTTPREL_LO12_NC
+      LDR   x2, :gottprel: symbol            // R_<CLS>_TLSIE_LD_GOTTPREL_PREL19
+
+      // after optimization:
+
+      NOP
+      MOV   x1, :tprel_g0: symbol            // R_<CLS>_TLSLE_MOVW_TPREL_G0
+      MOV   x2, :tprel_g0: symbol            // R_<CLS>_TLSLE_MOVW_TPREL_G0
+
+    Linkers must ensure that ``R_<CLS>_TLSIE_ADR_GOTTPREL_PAGE21`` is only optimized for a symbol if ``R_<CLS>_TLSIE_LD64_GOTTPREL_LO12_NC`` is optimized too.
```

Comment:
Could be worth emphasising the same ``symbol``. For example Linkers must ensure that ``R_<CLS>_TLSIE_ADR_GOTTPREL_PAGE21`` to ``symbol`` is only optimized if ``R_<CLS>_TLSIE_LD64_GOTTPREL_LO12_NC`` to ``symbol`` is also optimized.

Reply:
And it should be obvious it applies to all occurrences of these relocations. The statement is just to make it explicit there is a dependency rather than trying to describe the linker implementation in detail. I'll reword it slightly